var/spack/repos/builtin/packages/simmetrix-simmodsuite/package.py

```diff
@@ -0,0 +1,336 @@
+         'MeshSim': ['ea2246f55e1fcf732355afb832252b0c', 'base'],
+         'FieldSim': ['aeca95adfa92710bc6b731108124d961', 'base'],
+         'GeomSimParasolid': ['59680124b20a45fe3122f364be485398', 'parasolid'],
+         'GeomSimAbstract': ['dfe2559a7d1fb837a39afd9b24f12a26', 'abstract'],
+         'GeomSimDiscrete': ['233fa79ef98483ad69d45647ec409e48', 'discrete'],
+         'GeomSimProe': ['b2587521678eb121384d22d686b5456d', 'granite'],
+      }
+    }
+    ]
+    return releases
+
+
+def simmetrix_getKernel():
+    """only supporting the linux libraries"""
+    return "linux"
+
+
+def simmetrix_getWordSz():
+    """only supporting 64b libraries"""
+    return "64"
+
+
+def simmetrix_getLibDir(os):
+    osLibDir = {'rhel6': 'x64_rhel6_gcc44',
+                'rhel7': 'x64_rhel7_gcc48'}
+    return osLibDir[os]
+
+
+def simmetrix_makeComponentUrl(name):
+    """only supporting the linux libraries"""
+    prefix = "file://{0}/".format(os.getcwd())
+    suffix = "-" + simmetrix_getKernel() + simmetrix_getWordSz() + '.tgz'
+    return prefix + name + suffix
+
+
+def simmetrix_makeDocUrl(name):
+    """doc zip files are not os/arch specific"""
+    prefix = "file://{0}/".format(os.getcwd())
+    suffix = '.zip'
+    return prefix + name + suffix
+
+
+def simmetrix_setKernelCMakePrefixPath(spec, path, env):
+    if '+acis' in spec:
+        env.append_path('CMAKE_PREFIX_PATH', join_path(path, 'acisKrnl'))
+    if '+parasolid' in spec:
+        env.append_path('CMAKE_PREFIX_PATH', join_path(path, 'psKrnl'))
+
+
+def simmetrix_resource(name, url, md5, condition):
+    # The tarballs/zips each have the same directory structure.  Because of
+    # this, and the bug in spack described here:
+    # https://github.com/spack/spack/pull/3553#issuecomment-391424244
+    # , they cannot be expanded into the source root directory.
+    # Once this is fixed the 'destination=name' argument can be removed.
+    resource(
+      name=name,
+      url=url,
+      md5=md5,
+      destination=name,
+      when=condition
+    )
+
+
+def simmetrix_copytree(src, dst):
+    dir_util.copy_tree(src, dst)
+
+
+class SimmetrixSimmodsuite(Package):
+    """Simmetrix' Simulation Modeling Suite is a set of component software
+    toolkits that allow developers to easily implement geometry-based
+    simulation applications.
+    Each component of the Simulation Modeling Suite is designed to address
+    specific capabilities:
+    | MeshSim - automatic mesh generation
+    | FieldSim - simulation data management
+    | GeomSim - direct, untranslated access to geometry from a wide variety
+    of sources
+    """
+
+    homepage = "http://www.simmetrix.com/products/SimulationModelingSuite/main.html"
+
+    license_required = True
+    license_vars     = ['SIM_LICENSE_FILE']
+
+    variant('sim_os', default='rhel7', values=('rhel7', 'rhel6'),
+      description='installs libraries built on RedHat 6 or 7 Enterprise Linux')
```

Comment:
Let's get rid of this variant, Spack already detects the distro for you:
```python
if self.spec.satisfies('os=rhel6'):
    sim_os = 'rhel6'
elif self.spec.satisfies('os=rhel7'):
    sim_os = 'rhel7'
else:
    raise InstallError('Only RHEL 6 and 7 are supported')
```

Reply:
I've removed the OS variant and relaxed the check to look for the `linux` platform.  Historically the prepackaged builds have been portable among a handful of common linux distros (ubuntu, rhel, arch, debian, centos, etc.). I'm open to suggestions on better spack approaches to handle the platform dependency.